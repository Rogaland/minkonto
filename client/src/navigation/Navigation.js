// @flow weak

import React from 'react';
import AppBar from 'material-ui/AppBar';
import {withStyles} from 'material-ui/styles';
import PropTypes from 'prop-types';
import {Link} from "react-router-dom";
import {Tab, Tabs} from "material-ui";

const styles = theme => ({
    root: {
        marginTop: theme.spacing.unit * 3,
        width: '100%',
    },
});


class Navigation extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            value: 0
        };
    }


    handleChange = (event, value) => {
        this.setState({value});
    };

    render() {
        const {classes} = this.props;
        const {value} = this.state;


        return (
            <div className={classes.root}>
                <AppBar position="static">
                    <Tabs value={value} onChange={this.handleChange}>
                        <Tab label="Min profil" component={Link} to="/me"/>
                        <Tab label="Passord" component={Link} to="/password"/>
                    </Tabs>

                </AppBar>
            </div>
        )
    }
}

Navigation.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Navigation);

