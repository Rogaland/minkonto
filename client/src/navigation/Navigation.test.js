import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

import Navigation from './Navigation';

describe('Navigation', () => {
  
  it('Renders Navigation component', () => {
    const component = (
      <MemoryRouter>
        <Navigation classes={{}} />
      </MemoryRouter>
    );

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

});