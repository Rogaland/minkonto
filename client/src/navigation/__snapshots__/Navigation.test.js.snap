// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Renders Navigation component 1`] = `
<div
  className="Navigation-root-1"
>
  <header
    className="MuiPaper-root-9 MuiPaper-shadow4-15 MuiAppBar-root-2 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7"
  >
    <div
      className="MuiTabs-root-36"
    >
      <div
        className="MuiTabs-flexContainer-37"
      >
        <div
          className="MuiTabs-scrollingContainer-38 MuiTabs-fixed-39"
          onScroll={[Function]}
          role="tablist"
          style={
            Object {
              "marginBottom": 0,
            }
          }
        >
          <div
            className="MuiTabs-flexContainer-37"
          >
            <a
              aria-selected={true}
              className="MuiButtonBase-root-59 MuiTab-root-43 MuiTab-rootInherit-51 MuiTab-rootInheritSelected-52"
              disabled={false}
              href="/me"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              style={undefined}
              tabIndex={0}
            >
              <span
                className="MuiTab-wrapper-55"
              >
                <div
                  className="MuiTab-labelContainer-56"
                >
                  <span
                    className="MuiTab-label-57"
                  >
                    Min profil
                  </span>
                </div>
              </span>
              <span
                className="MuiTouchRipple-root-64"
              />
            </a>
            <a
              aria-selected={false}
              className="MuiButtonBase-root-59 MuiTab-root-43 MuiTab-rootInherit-51"
              disabled={false}
              href="/password"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              style={undefined}
              tabIndex={0}
            >
              <span
                className="MuiTab-wrapper-55"
              >
                <div
                  className="MuiTab-labelContainer-56"
                >
                  <span
                    className="MuiTab-label-57"
                  >
                    Passord
                  </span>
                </div>
              </span>
              <span
                className="MuiTouchRipple-root-64"
              />
            </a>
          </div>
          <div
            className="MuiTabIndicator-root-61 MuiTabIndicator-colorAccent-62"
            style={
              Object {
                "left": 0,
                "width": 0,
              }
            }
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
